### ISSUE TITLE
Different validation mAP when finetuning RFDETR Nano with `resolution=320|640` despite possible silent fallback to 384

### ISSUE THREAD

**Issue by Mhd-i** [roboflow/rf-detr#1001]
I fine-tuned two models from RFDETRNano using rfdetr 1.6.4, once with resolution=320 and once with resolution=640. from my understanding, version 1.6.4 silently ignores unsupported resolutions and falls back to 384 during training. If that is the case, I would expect both training runs to produce identical validation metrics, yet I observe markedly different EMA mAP values:

resolution=320 → best EMA mAP reaches 0.2560 by epoch 4

resolution=640 → best EMA mAP reaches 0.3192 by epoch 4

Full logs:
```
RFDETR Nano with resolution=320
[2026-04-27 15:12:41] [INFO] rf-detr - Best EMA mAP improved to 0.0014 (epoch 0)
[2026-04-27 15:14:07] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth
[2026-04-27 15:14:07] [INFO] rf-detr - Best EMA mAP improved to 0.0267 (epoch 0)
[2026-04-27 15:16:25] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 1)
[2026-04-27 15:16:26] [INFO] rf-detr - Best EMA mAP improved to 0.0829 (epoch 1)
[2026-04-27 15:19:03] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 2)
[2026-04-27 15:19:03] [INFO] rf-detr - Best EMA mAP improved to 0.1474 (epoch 2)
[2026-04-27 15:21:43] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 3)
[2026-04-27 15:21:43] [INFO] rf-detr - Best EMA mAP improved to 0.2030 (epoch 3)
[2026-04-27 15:24:29] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 4)
[2026-04-27 15:24:29] [INFO] rf-detr - Best EMA mAP improved to 0.2560 (epoch 4)
```
RFDETR Nano with resolution=640
```
[2026-04-27 15:33:26] [INFO] rf-detr - Best EMA mAP improved to 0.0006 (epoch 0)
[2026-04-27 15:36:13] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 0)
[2026-04-27 15:36:13] [INFO] rf-detr - Best EMA mAP improved to 0.0490 (epoch 0)
[2026-04-27 15:39:45] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 1)
[2026-04-27 15:39:45] [INFO] rf-detr - Best EMA mAP improved to 0.1600 (epoch 1)
[2026-04-27 15:43:48] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 2)
[2026-04-27 15:43:48] [INFO] rf-detr - Best EMA mAP improved to 0.2631 (epoch 2)
[2026-04-27 15:47:53] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 3)
[2026-04-27 15:47:54] [INFO] rf-detr - Best EMA mAP improved to 0.2928 (epoch 3)
[2026-04-27 15:51:47] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 4)
[2026-04-27 15:51:48] [INFO] rf-detr - Best EMA mAP improved to 0.3192 (epoch 4)
```

**Comment by Mhd-i** [comment]
here is another example, both models are finetuned on the same dataset and same seed:

RFDETR Nano resolution=640 seed=42
```
[2026-04-27 19:33:16] [INFO] rf-detr - Best EMA mAP improved to 0.0006 (epoch 0)
[2026-04-27 19:36:12] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 0)
[2026-04-27 19:36:12] [INFO] rf-detr - Best EMA mAP improved to 0.0507 (epoch 0)
[2026-04-27 19:39:41] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 1)
[2026-04-27 19:39:41] [INFO] rf-detr - Best EMA mAP improved to 0.1390 (epoch 1)
[2026-04-27 19:43:32] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 2)
[2026-04-27 19:43:32] [INFO] rf-detr - Best EMA mAP improved to 0.2072 (epoch 2)
[2026-04-27 19:47:25] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 3)
[2026-04-27 19:47:25] [INFO] rf-detr - Best EMA mAP improved to 0.2796 (epoch 3)
[2026-04-27 19:51:09] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 4)
[2026-04-27 19:51:10] [INFO] rf-detr - Best EMA mAP improved to 0.3258 (epoch 4)
[2026-04-27 19:55:03] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 5)
[2026-04-27 19:55:03] [INFO] rf-detr - Best EMA mAP improved to 0.3805 (epoch 5)
[2026-04-27 19:58:56] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 6)
[2026-04-27 19:58:56] [INFO] rf-detr - Best EMA mAP improved to 0.3966 (epoch 6)
[2026-04-27 20:03:09] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 7)
[2026-04-27 20:03:10] [INFO] rf-detr - Best EMA mAP improved to 0.4265 (epoch 7)
[2026-04-27 20:06:57] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 8)
[2026-04-27 20:06:57] [INFO] rf-detr - Best EMA mAP improved to 0.4339 (epoch 8)
[2026-04-27 20:11:12] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 9)
[2026-04-27 20:11:12] [INFO] rf-detr - Best EMA mAP improved to 0.4533 (epoch 9)
[2026-04-27 20:15:15] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 10)
[2026-04-27 20:15:15] [INFO] rf-detr - Best EMA mAP improved to 0.4600 (epoch 10)
```
RFDETR Nano resolution=320 seed=42
```
[2026-04-27 20:36:28] [INFO] rf-detr - Best EMA mAP improved to 0.0014 (epoch 0)
[2026-04-27 20:37:55] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 0)
[2026-04-27 20:37:55] [INFO] rf-detr - Best EMA mAP improved to 0.0328 (epoch 0)
[2026-04-27 20:40:29] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 1)
[2026-04-27 20:40:29] [INFO] rf-detr - Best EMA mAP improved to 0.0897 (epoch 1)
[2026-04-27 20:43:39] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 2)
[2026-04-27 20:43:39] [INFO] rf-detr - Best EMA mAP improved to 0.1618 (epoch 2)
[2026-04-27 20:46:47] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 3)
[2026-04-27 20:46:47] [INFO] rf-detr - Best EMA mAP improved to 0.2357 (epoch 3)
[2026-04-27 20:50:10] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 4)
[2026-04-27 20:50:11] [INFO] rf-detr - Best EMA mAP improved to 0.2683 (epoch 4)
[2026-04-27 20:53:02] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 5)
[2026-04-27 20:53:03] [INFO] rf-detr - Best EMA mAP improved to 0.3149 (epoch 5)
[2026-04-27 20:56:26] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 6)
[2026-04-27 20:56:27] [INFO] rf-detr - Best EMA mAP improved to 0.3295 (epoch 6)
[2026-04-27 20:59:58] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 7)
[2026-04-27 20:59:58] [INFO] rf-detr - Best EMA mAP improved to 0.3427 (epoch 7)
[2026-04-27 21:06:22] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 9)
[2026-04-27 21:06:23] [INFO] rf-detr - Best EMA mAP improved to 0.3484 (epoch 9)
[2026-04-27 21:09:30] [INFO] rf-detr - Best regular mAP saved to /kaggle/working/checkpoint_best_regular.pth (epoch 10)
[2026-04-27 21:09:30] [INFO] rf-detr - Best EMA mAP improved to 0.3602 (epoch 10)
```
